tools/sigclient/pkg/query/validation.go

```diff
@@ -597,36 +602,60 @@ func ValidateStatsQueryResults(queryRes *Result, expRes *Result) error {
 	sortMeasureResults(queryRes.MeasureResults)
 	sortMeasureResults(expRes.MeasureResults)
 
+	verifyGroupByValues := !expRes.DoNotVerifyGroupByValues
+
+	if !verifyGroupByValues {
+		log.Infof("ValidateStatsQueryResults: Skipping GroupByValues Elements match.")
+	}
+
 	for idx, expMeasureRes := range expRes.MeasureResults {
 		var err error
 		queryMeasureRes := queryRes.MeasureResults[idx]
 
-		equal := reflect.DeepEqual(queryMeasureRes.GroupByValues, expMeasureRes.GroupByValues)
-		if !equal {
-			return fmt.Errorf("ValidateStatsQueryResults: GroupByCombination mismatch, expected: %+v, got: %+v", expMeasureRes.GroupByValues, queryMeasureRes.GroupByValues)
+		if len(queryMeasureRes.GroupByValues) != len(expMeasureRes.GroupByValues) {
```

Comment:
Do we need the length check here? Wouldn't reflect.DeepEqual check that?

Reply:
yes it will check, but I added this as we want to check the length of the `groupByvalues` slice when we do not verify them.
may be I will add that check in the else condition.